### PR TITLE
Sastai develope

### PR DESCRIPTION
**Pull request by nilsnevertree**
Partly closes issue #67
- [x] Index implementation
- [x] Testing
- [x] Choose between ocean, land only or full hemisphere not yet included. Can be done after merge #87 is performed.

### CODE REVIEW

**Comment by willirath** [comment]
@nilsnevertree #87 is merged.

**Comment by nilsnevertree** [comment]
New indices and tests are implemented and the branch is now ready to be merged into main.
Could one of you review the pull request @willirath or @MarcoLandtHayen ?

Please note, that still #95 exists and needs to be closed by #96

**Comment by willirath** [comment]
I've merged #96 and updated this branch my doing
```shell
$ git checkout sastai-develope
$ git pull
$ git merge origin/main
$ git push
```

File: climate_index_collection/indices.py [hunk 1]
```diff
@@ -320,6 +325,263 @@ def north_atlantic_sea_surface_salinity(data_set, sss_name="sea-surface-salinity
...
+# SASTAI north
+
+
+def sea_air_surface_temperature_anomaly_north_all(
+    data_set, sat_name="sea-air-temperature"
+):
+    """Sea Air Surface Temperature Anomaly (SASTA) index ,
+    for the northern hemisphere.
+    Land and Ocean data is used for the calculation.
+    The Anomalies are climatoligical anomalies (monthly) relative to the whole time period of the data_set.
```
Comment (willirath) [inline climate_index_collection/indices.py:342]:
```suggestion
    """Calculate the Sea Air Surface Temperature Anomaly (SASTA) index, for the northern hemisphere.
    
    Land and Ocean data is used for the calculation. The anomalies are relative to a monthly climatology
    calculated from the whole time covered by the data set.
```

File: climate_index_collection/indices.py [hunk 2]
```diff
@@ -320,6 +325,263 @@ def north_atlantic_sea_surface_salinity(data_set, sss_name="sea-surface-salinity
...
+    return SASTAI
+
+
+def sea_air_surface_temperature_anomaly_north_ocean(
+    data_set, sat_name="sea-air-temperature"
+):
+    """Sea Air Surface Temperature Anomaly (SASTA) index ,
+    for the northern hemisphere.
+    Only data over the Ocean is used for the calculation.
+    The Anomalies are climatoligical anomalies (monthly) relative to the whole time period of the data_set.
```
Comment (willirath) [inline climate_index_collection/indices.py:383]:
```suggestion
    """Calculate the Sea Air Surface Temperature Anomaly (SASTA) index, for the northern hemisphere.
    
    Only data over the Ocean is used for the calculation. The anomalies are relative to a monthly climatology
    calculated from the whole time covered by the data set.
```

File: climate_index_collection/indices.py [hunk 3]
```diff
@@ -320,6 +325,263 @@ def north_atlantic_sea_surface_salinity(data_set, sss_name="sea-surface-salinity
...
+    return SASTAI
+
+
+def sea_air_surface_temperature_anomaly_north_land(
+    data_set, sat_name="sea-air-temperature"
+):
+    """Sea Air Surface Temperature Anomaly (SASTA) index ,
+    for the northern hemisphere.
+    Only data over land is used for the calculation.
+    The Anomalies are climatoligical anomalies (monthly) relative to the whole time period of the data_set.
```
Comment (willirath) [inline climate_index_collection/indices.py:424]:
```suggestion
    """Calculate the Sea Air Surface Temperature Anomaly (SASTA) index, for the northern hemisphere.
    
    Only data over land is used for the calculation. The anomalies are relative to a monthly climatology
    calculated from the whole time covered by the data set.
```

File: climate_index_collection/indices.py [hunk 4]
```diff
@@ -320,6 +325,263 @@ def north_atlantic_sea_surface_salinity(data_set, sss_name="sea-surface-salinity
...
+# SASTAI south
+
+
+def sea_air_surface_temperature_anomaly_south_all(
+    data_set, sat_name="sea-air-temperature"
+):
+    """Sea Air Surface Temperature Anomaly (SASTA) index ,
+    for the southern hemisphere.
+    Land and Ocean data is used for the calculation.
+    The Anomalies are climatoligical anomalies (monthly) relative to the whole time period of the data_set.
```
Comment (willirath) [inline climate_index_collection/indices.py:468]:
```suggestion
    """Calculate the Sea Air Surface Temperature Anomaly (SASTA) index, for the southern hemisphere.
    
    Land and Ocean data is used for the calculation. The anomalies are relative to a monthly climatology
    calculated from the whole time covered by the data set.
```

File: climate_index_collection/indices.py [hunk 5]
```diff
@@ -320,6 +325,263 @@ def north_atlantic_sea_surface_salinity(data_set, sss_name="sea-surface-salinity
...
+    return SASTAI
+
+
+def sea_air_surface_temperature_anomaly_south_ocean(
+    data_set, sat_name="sea-air-temperature"
+):
+    """Sea Air Surface Temperature Anomaly (SASTA) index ,
+    for the southern hemisphere.
+    Only data over the Ocean is used for the calculation.
+    The Anomalies are climatoligical anomalies (monthly) relative to the whole time period of the data_set.
```
Comment (willirath) [inline climate_index_collection/indices.py:509]:
```suggestion
    """Calculate the Sea Air Surface Temperature Anomaly (SASTA) index, for the southern hemisphere.
    
    Only data over the Ocean is used for the calculation. The anomalies are relative to a monthly climatology
    calculated from the whole time covered by the data set.
```

File: climate_index_collection/indices.py [hunk 6]
```diff
@@ -320,6 +325,263 @@ def north_atlantic_sea_surface_salinity(data_set, sss_name="sea-surface-salinity
...
+    return SASTAI
+
+
+def sea_air_surface_temperature_anomaly_south_land(
+    data_set, sat_name="sea-air-temperature"
+):
+    """Sea Air Surface Temperature Anomaly (SASTA) index ,
+    for the southern hemisphere.
+    Only data over land is used for the calculation.
+    The Anomalies are climatoligical anomalies (monthly) relative to the whole time period of the data_set.
```
Comment (willirath) [inline climate_index_collection/indices.py:550]:
```suggestion
    """Calculate the Sea Air Surface Temperature Anomaly (SASTA) index, for the southern hemisphere.
    
    Only data over land is used for the calculation. The anomalies are relative to a monthly climatology
    calculated from the whole time covered by the data set.
```

File: tests/test_indices.py [hunk 7]
```diff
@@ -13,275 +13,538 @@
...
+#     # Check, if calculated SAM index only has one dimension: 'time'
+#     assert SAM.dims[0] == "time"
+#     assert len(SAM.dims) == 1
+
+
+# @pytest.mark.parametrize("source_name", list(VARNAME_MAPPING.keys()))
+# def test_SAM_standardisation(source_name):
+#     """Ensure that standardisation works correctly."""
+#     # Load test data
+#     TEST_DATA_PATH = Path(__file__).parent / "../data/test_data/"
```
Comment (willirath) [inline tests/test_indices.py:46]: I don't understand why there's so much code commented out here.

**Review by willirath — CHANGES_REQUESTED** [review]
I've left suggestions for the docstrings.

And I think the code that was commented out in the tests should be re-activated.

File: tests/test_indices.py (hunk 7 above)
Comment (nilsnevertree) [reply tests/test_indices.py:46]: Ah my bad, missed that and will correct it.
Just commented it out for the tests to run faster.
And the docstring is much better.

Comment (nilsnevertree) [reply tests/test_indices.py]: Resolved with commit 9cbb69f0a3c925cabb885f5452ed5b7a77b630a2

File: tests/test_indices.py [hunk 8]
```diff
@@ -285,3 +291,260 @@ def test_NASSS_standardisation(source_name):
...
+    # Load test data
+    TEST_DATA_PATH = Path(__file__).parent / "../data/test_data/"
+    data_set = load_data_set(data_path=TEST_DATA_PATH, data_source_name=source_name)
+
+    # Calculate SASTAI
+    SASTAI = sea_air_surface_temperature_anomaly_north_all(data_set)
+
+    # Check, if calculated SASTAI only has one dimension: 'time'
+    assert SASTAI.dims[0] == "time"
+    assert len(SASTAI.dims) == 1
```
Comment (willirath) [inline tests/test_indices.py:316]: And I'd parameterize over the different SASTAI index functions here.

**Review by willirath — CHANGES_REQUESTED** [review]
I left a few comments.

As there's also a few typos in the docstrings, I'll just add a commit on my own implementing these changes.

**Review by willirath — APPROVED** [review]

**Pull request merged**